node/src/components/linear_chain.rs

```diff
@@ -47,12 +48,17 @@ pub enum Event<I> {
     #[from]
     Request(LinearChainRequest<I>),
     /// New linear chain block has been produced.
-    LinearChainBlock {
+    NewLinearChainBlock {
         /// The block.
         block: Box<Block>,
         /// The deploys' execution results.
         execution_results: HashMap<DeployHash, ExecutionResult>,
     },
+    /// Linear chain block we already know but we may refinalize it when syncing protocol state.
+    KnownLinearChainBlock {
+        /// The block.
+        block: Box<Block>,
+    },
```

Comment:
Maybe a tuple struct variant would be more concise?
```suggestion
    KnownLinearChainBlock(Box<Block>),
```

Reply:
Done in [`31bccd8` (#1144)](https://github.com/CasperLabs/casper-node/pull/1144/commits/31bccd8eacaf40767367af900ea6e25019463f0f)